tiledb/sm/array_schema/array_schema.h

```diff
@@ -102,19 +103,31 @@ class ArraySchema {
    * Returns a constant pointer to the selected attribute (nullptr if it
    * does not exist).
    */
-  const Attribute* attribute(unsigned int id) const;
+  shared_ptr<const Attribute> attribute(unsigned int id) const;
 
   /**
    * Returns a constant pointer to the selected attribute (nullptr if it
    * does not exist).
    */
-  const Attribute* attribute(const std::string& name) const;
+  Attribute* attribute(const std::string& name) const;
```

Comment:
Why is the `const` qualifier dropped here? If users of this class are already using constant pointers, it's better to keep it that way.